src/panel.rs

```diff
@@ -144,23 +144,23 @@ impl<K: Hash + Eq + Copy + Display + 'static> TabPanel<K> {
     /// It is fail-safe, if the postion is greater than the amount of tabs, it is appended to the end.
     /// Note: Calls `add_tab_at` on the enclosed `TabView`.
     pub fn add_tab_at<T: View>(&mut self, id: K, view: T, pos: usize) {
-        self.tabs.add_tab_at(id, view, pos);
+        self.tabs.add_tab_at(id.clone(), view, pos);
         self.bar.add_button_at(self.tx.clone(), id, pos);
     }
 
     /// Consuming & Chainable variant to add a new tab at a certain position.
     /// It is fail-safe, if the postion is greater than the amount of tabs, it is appended to the end.
     /// Note: Calls `add_tab_at` on the enclosed `TabView`.
     pub fn with_tab_at<T: View>(mut self, id: K, view: T, pos: usize) -> Self {
-        self.tabs.add_tab_at(id, view, pos);
+        self.tabs.add_tab_at(id.clone(), view, pos);
         self.bar.add_button_at(self.tx.clone(), id, pos);
         self
     }
 
     /// Remove a tab of the enclosed `TabView`.
     pub fn remove_tab(&mut self, id: K) -> Result<(), ()> {
-        self.bar.remove_button(id);
-        self.tabs.remove_tab(id)
+        self.bar.remove_button(&id);
+        self.tabs.remove_tab(&id)
```

Comment:
I would say let's simply pass the reference to the function itself, this way we also behave similar as tab view already does.
So change the definition to something like this:
```rust
pub fn remove_tab(&mut self, id: &K) -> Result<(),()> {
```